pkg/config/configuration.go

```diff
@@ -336,9 +341,21 @@ func LoadKubernetesConfiguration(config, namespace string, podName string, opera
 		return nil, err
 	}
 
+	sortMetricsSpec(conf)
 	return conf, nil
 }
 
+// Apply .metrics if set. If not, retain .metric.
+func sortMetricsSpec(conf *Configuration) {
+	if !conf.Spec.MetricsSpec.Enabled {
```

Comment:
I think that if we use .metrics for one thing, we should use it for all. With this code, a user could specify .metrics for one thing (for example, enabled), and .metric for the other (rules).

Reply:
The issue is that both metric and metrics are set to true by default, so we can't know if one field was set unless it's false, in which case all other settings are ignored. It's unlikely someone would use both, but if they do, it should be legal for as long as we support both I think.